.github/workflows/build-package-docs.yaml

```diff
@@ -120,3 +132,77 @@ jobs:
         name: package-docs-build
         path: build-directory/docs/docsite/ansible-package-docs-html-*.tar.gz
         retention-days: 7
+
+  deploy-package-docs:
+    if: fromJSON(github.event.inputs.deploy)
+    needs: build-package-docs
+    runs-on: ubuntu-latest
+    environment:
+      name: deploy-package-docs
+      url: https://github.com/ansible/ansible-documentation
+    env:
+      dest-repo: owner/repo-name  # TODO create repo and specify here
+      dest-branch: main  # TODO specify repo branch here
+      user-email: "41898282+github-actions[bot]@users.noreply.github.com"
+      user-name: "github-actions[bot]"
+    steps:
+    - name: Download the build archive
+      uses: actions/download-artifact@v4
+      with:
+        name: package-docs-build
+
+    - name: Extract the tarball
+      run: |
+        set -xeEuo pipefail
+        mkdir -pv extracted-docs
+        tar -xvzf ansible-package-docs-html-*.tar.gz -C extracted-docs
+
+    - name: Generate temp GITHUB_TOKEN
+      id: create_token
+      uses: tibdex/github-app-token@v2
+      with:
+        app_id: ${{ secrets.BOT_APP_ID }}  # TODO create bot app id secret
+        private_key: ${{ secrets.BOT_APP_KEY }}  # TODO create bot app id key
+
+    - name: Checkout the deploy directory
+      uses: actions/checkout@v4
+      with:
+        repository: ${{ env.dest-repo }}
+        ref: ${{ env.dest-branch }}
+        path: deploy-directory
+        token: ${{ steps.create_token.outputs.token }}
+
+    - name: Copy the generated HTML and assets for production
+      if: ${{ github.event.inputs.deployment-environment == 'production' }}
+      run: >-
+        rsync -av --delete --mkpath
+        extracted-docs/
+        deploy-directory/docs-build
+
+    - name: Add generated HTML and assets
+      run: |
+        git config --local user.email "${{ env.user-email }}"
+        git config --local user.name "${{ env.user-name }}"
+        git add . --all --force
+      working-directory: deploy-directory
+
+    - name: Commit generated HTML and assets
+      run: >-
+        git diff-index --quiet HEAD ||
+        git commit -m "Push docs build $(date '+%Y-%m-%d')-${{
+          github.run_id
+        }}-${{
+           github.run_number
+        }}-${{
+          github.run_attempt
+        }}"
+      working-directory: deploy-directory
+
+    - name: Push build to deploy repository
+      uses: ad-m/github-push-action@v0.8.0
+      with:
+        github_token: ${{ steps.create_token.outputs.token }}
+        repository: ${{ env.dest-repo }}
+        branch: ${{ env.dest-branch }}
+        directory: deploy-directory
+        force: true
```

Comment:
Does this action do something particularly special that plain `git push` wouldn't accomplish? Generally, I prefer to avoid external GHA dependencies for things like this.

Reply:
I don't think it does anything particularly special over a plain `git push`. I do feel like it makes the workflow a little more readable but it's not a strong preference.

Reply:
Plain `git push` seems to work just fine with a deploy key so let's keep it simple.